middleware/packet-forward-middleware/docs/integration.md

```diff
@@ -0,0 +1,156 @@
+# Integration
+This document provides instructions on integrating and configuring the Packet Forward Middleware (PFM) within your
+existing chain implementation. This document is _NOT_ a guide on developing with the Cosmos SDK or ibc-go and makes
+the assumption that you have some existing codebase for your chain with IBC already enabled.
+
+## Example integration of the Packet Forward Middleware
+
+```go
+// app.go
+
+// Import the packet forward middleware
+import (
+    packetforward "github.com/cosmos/ibc-apps/middleware/packet-forward-middleware/v7/router"
+    packetforwardkeeper "github.com/cosmos/ibc-apps/middleware/packet-forward-middleware/v7/router/keeper"
+    packetforwardtypes "github.com/cosmos/ibc-apps/middleware/packet-forward-middleware/v7/router/types"
+)
+
+...
+
+// Register the AppModule for the packet forward middleware module
+ModuleBasics = module.NewBasicManager(
+    ...
+    packetforward.AppModuleBasic{},
+    ...
+)
+
+...
+
+// Add packet forward middleware Keeper
+type App struct {
+	...
+	PacketForwardKeeper *packetforwardkeeper.Keeper
+	...
+}
+
+...
+
+// Create store keys 
+keys := sdk.NewKVStoreKeys(
+    ...
+    packetforwardtypes.StoreKey,
+    ...
+)
+
+...
+
+// Initialize the packet forward middleware Keeper
+// It's important to note that the PFM Keeper must be initialized before the Transfer Keeper
+app.PacketForwardKeeper = packetforwardkeeper.NewKeeper(
+    appCodec,
+    keys[packetforwardtypes.StoreKey],
+    app.GetSubspace(packetforwardtypes.ModuleName),
+    app.TransferKeeper, // will be zero-value here, reference is set later on with SetTransferKeeper.
+    app.IBCKeeper.ChannelKeeper,
+    appKeepers.DistrKeeper,
+    app.BankKeeper,
+    app.IBCKeeper.ChannelKeeper,
+)
+
+// Initialize the transfer module Keeper
+app.TransferKeeper = ibctransferkeeper.NewKeeper(
+    appCodec,
+    keys[ibctransfertypes.StoreKey],
+    app.GetSubspace(ibctransfertypes.ModuleName),
+    app.PacketForwardKeeper,
+    app.IBCKeeper.ChannelKeeper,
+    &app.IBCKeeper.PortKeeper,
+    app.AccountKeeper,
+    app.BankKeeper,
+    scopedTransferKeeper,
+)
+
+app.PacketForwardKeeper.SetTransferKeeper(app.TransferKeeper)
+
+// See the section below for configuring an application stack with the packet forward middleware 
+
+...
+
+// Register packet forward middleware AppModule
+app.moduleManager = module.NewManager(
+    ...
+    packetforward.NewAppModule(app.PacketForwardKeeper),
+)
+
+...
+
+// Add packet forward middleware to begin blocker logic
+app.moduleManager.SetOrderBeginBlockers(
+    ...
+    packetforwardtypes.ModuleName,
+    ...
+)
+
+// Add packet forward middleware to end blocker logic
+app.moduleManager.SetOrderEndBlockers(
+    ...
+    packetforwardtypes.ModuleName,
+    ...
+)
+
+// Add packet forward middleware to init genesis logic
+app.moduleManager.SetOrderInitGenesis(
+    ...
+    ibcfeetypes.ModuleName,
```

Comment:
Is this right?

Reply:
as in, should it even be included in `SetOrderInitGenesis` at all? I looked at another PR for integrating the PFM and I saw this included, I do see logic in `keeper.InitGenesis` in the PFM as well so I figured that made sense.